### PR TITLE
feat: Migrate ms-ledger microservice to EdgeX 3.0

### PR DESCRIPTION
**Pull request by NeethuES-intel**
## PR Checklist

- [X] Added label to the Pull Request for easier discoverability and search
- [X] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/automated-vending/blob/main/.github/CONTRIBUTING.md
- [X] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [X] All commented out code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## Issue this PR will close

close: #131

## Anything the reviewer should know when reviewing this PR?
PS: I was unable to create branch directly from the repo, hence had to fork.

## Test Instructions if applicable
Run the following make commands from the ms-ledger directory
- make docker
- make test

### CODE REVIEW

File: ms-ledger/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 
-FROM golang:1.18-alpine3.16 AS builder
+FROM golang:1.20-alpine3.17 AS builder
 
 LABEL license='SPDX-License-Identifier: BSD-3-Clause' \
-  copyright='Copyright (c) 2022: Intel'
+  copyright='Copyright (c) 2023: Intel'
 
 
 # add git for go modules
```
Comment (lenny-intel) [inline ms-ledger/Dockerfile:10]:
```suggestion
# add git for go modules
# hadolint ignore=DL3018
RUN apk update && apk add --no-cache make git
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by lenny-intel — COMMENTED** [review]
Need to remove `-tags no_zmq` from test targets in make file

File: ms-ledger/go.mod [hunk 2]
```diff
@@ -6,69 +6,73 @@ module ms-ledger
 go 1.18
```
Comment (vli11) [inline ms-ledger/go.mod:6]: this needs to be go 1.20

**Review by lenny-intel — APPROVED** [review]
LGTM, just need to update service name in .github/workflows/trivyimagescan.yaml to resolve last failing check.

**Review by lenny-intel — APPROVED** [review]

**Review by brian-intel — APPROVED** [review]

**Pull request merged**